### PR TITLE
refactoring: split up and simplify runTCPCheck

### PR DESCRIPTION
**Pull request by soupdiver**
Added some comments, split up the function, reversed some `if` logic to exit early and removed named return values for consistency.
Please check the TODO if the case is handled properly.

Cyclomatic complexity reduced from 21 to 5

### CODE REVIEW

File: tcp.go [hunk 1]
```diff
@@ -103,20 +122,13 @@ func (fm *Frontman) runTCPCheck(addr *net.TCPAddr, hostname string, service stri
 		err = checkHTTPS(conn, hostname, secToDuration(fm.NetTCPTimeout))
 		break
 	case "tcp":
+		// TODO: Nothing to do in tcp case or report an error? If nothing why?
```
Comment (requilence) [inline tcp.go]: Because we just need to test the TCP connection on port in this case

Comment (soupdiver) [reply tcp.go]: So a comment would be nice in this case

**Comment by soupdiver** [comment]
@requilence What about this line: https://github.com/cloudradar-monitoring/frontman/pull/14/files#diff-a6af1450c8e7f80ac48e4d80e5401949R43

Is it ok to return `nil` here or should it also be the `MeasurementsMap`. Because it's just initialised afterwards.

**Comment by requilence** [comment]
I remember we talked with @thorstenkramm about this case. If the data is incorrect (can't determine port by service) and the check can't be performed we should return nil. BUT. I just noticed that I forgot one important thing here, that I was done in ssl.go – check the OS repo for the service-port mappings https://github.com/cloudradar-monitoring/frontman/blob/master/ssl.go#L29

**Comment by soupdiver** [comment]
> I just noticed that I forgot one important thing here, that I was done in ssl.go – check the OS repo for the service-port mappings

So we don't need our own mapping at all?

**Comment by requilence** [comment]
> > I just noticed that I forgot one important thing here, that I was done in ssl.go – check the OS repo for the service-port mappings
> 
> So we don't need our own mapping at all?

We still need as a fallback (e.g. for Windows). Golang's net lib has it's own fallback but it doesn't contains some of services we use(ldap,nntp)

**Comment by soupdiver** [comment]
> We still need as a fallback (e.g. for Windows). Golang's net lib has it's own fallback but it doesn't contains some of services we use(ldap,nntp)

gotcha

**Comment by soupdiver** [comment]
> Should I add the missing port lookup by myself?
Let's do this in a separate PR? Then this one contains only the refactoring.
You review my changes. If everything is approved by you feel free to merge

File: tcp.go [hunk 2]
```diff
@@ -32,33 +32,55 @@ var defaultPortByService = map[string]int{
...
+	}
+
+	// Mark check as successfull
+	m[prefix+"success"] = 1
+
+	return m, nil
+}
+
+// executeCheck executes the check based on the passed service name
+func (fm *Frontman) executeCheck(conn net.Conn, service, hostname string) error {
```
Comment (requilence) [inline tcp.go]: One more thing.  Frontman is the top-level entity and shared across different types of checks: ICMP, SSL, TCP, Web.

So now you split it into the 2 top-level methods.
1. I think you can remove the struct pointer here and just pass `fm.NetTCPTimeout` as arg
2. You should rename the method it to mention the type of check. Maybe  executeTCPCheck

Comment (soupdiver) [reply tcp.go]: 1. ✅ 
2. The function does not plainly check TCP. Inside the method there are differentiations which protocol is checked. The connection is also given to the function. Therefor I think adding TCP to the name could be a bit misleading 🤔

Comment (requilence) [reply tcp.go]: TCP is a transport protocol the other ones are underlying protocols. So you can use smth like "checkUnderlyingTCProtocol"

**Pull request merged**